src/main/java/org/broadinstitute/barclay/argparser/CommandLineArgumentParser.java

```diff
@@ -1080,6 +1081,11 @@ public ArgumentDefinition(
                     throw new CommandLineException.CommandLineParserInternalException(String.format("Integer argument --%s has a minimum or maximum attribute with a non-integral value.", this.getLongName()));
                 }
             }
+            this.isHidden = field.getAnnotation(Hidden.class) != null;
+            if (this.isHidden && !this.optional) {
+                // required arguments cannot be optional, because they should be provided in the command line
+                throw new CommandLineException.CommandLineParserInternalException(String.format("Required argument --%s could not be hidden for the command line", this.getLongName()));
```

Comment:
Good catch. Can you change this string to say something like "A required argument cannot be annotated with @Hidden: %s".